markets/storageadapter/getcurrentdealinfo.go

```diff
@@ -8,72 +8,107 @@ import (
 	"github.com/filecoin-project/go-state-types/abi"
 	"github.com/filecoin-project/go-state-types/exitcode"
 	"github.com/filecoin-project/lotus/api"
+	"github.com/filecoin-project/lotus/build"
 	"github.com/filecoin-project/lotus/chain/actors/builtin/market"
 	"github.com/filecoin-project/lotus/chain/types"
+	market2 "github.com/filecoin-project/specs-actors/v2/actors/builtin/market"
 	"github.com/ipfs/go-cid"
 	"golang.org/x/xerrors"
 )
 
 type getCurrentDealInfoAPI interface {
+	ChainGetMessage(ctx context.Context, mc cid.Cid) (*types.Message, error)
 	StateLookupID(context.Context, address.Address, types.TipSetKey) (address.Address, error)
 	StateMarketStorageDeal(context.Context, abi.DealID, types.TipSetKey) (*api.MarketDeal, error)
-	StateSearchMsg(context.Context, cid.Cid) (*api.MsgLookup, error)
+	StateWaitMsg(ctx context.Context, msg cid.Cid, confidence uint64) (*api.MsgLookup, error)
```

Comment:
For some reason when I use StateSearchMsg, in some of the integration tests StateSearchMsg would return an error indicating the [block was not found in the blockstore](https://app.circleci.com/pipelines/github/filecoin-project/lotus/12358/workflows/006e0e18-a7a7-4b51-a93d-d38a868d9c58/jobs/105721). But when I used StateWaitMsg I don't get this error.
This is strange because before we get to calling GetCurrentDealInfo we have already called StateWaitMsg, so the message should have been executed successfully.